SLING-11150: Return package id in the DistributionResponse for a dete…

amit-jain opened the pull request:
…rministic way of identifying status

- Add DistributionResponseInfo object to DistributionResponse

Commit:
SLING-11150: Return package id in the DistributionResponse for a deterministic way of identifying status

- Add DistributionResponseInfo object to DistributionResponse

amit-jain commented:
@tmaret Could you please review. Thanks!

amit-jain committed:
SLING-11150: Return package id in the DistributionResponse for a deterministic way of identifying status

- Test for default method in DistributionResponse

tmaret commented on src/main/java/org/apache/sling/distribution/DistributionResponse.java:
```diff
@@ -57,4 +57,13 @@
+     * returns additional properties related to the {@link DistributionRequest}
+     * 
+     * @return additional properties
+     */
```
If I understand the intention properly, this method should be annotated with `@Nonnull`

amit-jain committed:
GRANITE-37535: Return a unique identifier for the package after the http replicate call in tests

Addressing review changes
- Remove the default modifier and add Nonnull annotation
- Remove the test for the default

amit-jain committed:
GRANITE-37535: Return a unique identifier for the package after the http replicate call in tests

Addressing review changes
- Remove the default modifier
- Modify the static instance NONE to return empty string

tmaret commented on src/test/java/org/apache/sling/distribution/DistributionResponseTest.java:
```diff
@@ -0,0 +1,92 @@
+            return message;
+        }
+        
+        @Override public DistributionResponseInfo getDistributionInfo() {
```
Should be annotated `@Nonnull`

tmaret reviewed (commented):
The api module is using Java 8 so it's technically possible to use default methods. But I think we should limit our use of `default` for ConsumerType to avoid breaking consumers. In this case we provide all the implementations and would be fine without providing a default implementation in the interface.

amit-jain committed:
GRANITE-37535: Return a unique identifier for the package after the http replicate call in tests

Addressing review changes
- Adding @Nonnull annotation to #getDistributionInfo()

tmaret reviewed (approved):
LGTM
